sdk/nodejs/apigateway/experimental/api.ts

```diff
@@ -399,9 +399,18 @@ function addSwaggerOperation(swagger: SwaggerSpec, path: string, method: string,
     swagger.paths[path][method] = operation;
 }
 
+function checkRoute<TRoute>(route: TRoute, propName: keyof TRoute, opts: pulumi.ResourceOptions) {
```

Comment:
using `propName: keyof TRoute` is a cute trick that ensures that hte string literals we pass in are known property names for the typeof route we're checing.  that way we don't rename one, and forget to update the checkRoute calls.

Reply:
nice!